modules/clickhouse/zookeeper.go

```diff
@@ -0,0 +1,129 @@
+package clickhouse
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/docker/go-connections/nat"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const defaultZKImage = "zookeeper:3.7"
+
+const zkPort = nat.Port("2181/tcp")
+
+// ClickHouseContainer represents the ClickHouse container type used in the module
+type ZookeeperContainer struct {
+	testcontainers.Container
+	exposedPort nat.Port
+	ipaddr      string
+}
+
+func (c *ZookeeperContainer) IP() string {
+	return c.ipaddr
+}
+
+func (c *ZookeeperContainer) Port() nat.Port {
+	return c.exposedPort
+}
+
+func (c *ZookeeperContainer) ClickHouseConfig() string {
+	return fmt.Sprintf(`<?xml version="1.0"?>
+<clickhouse>
+    <logger>
+        <level>debug</level>
+        <console>true</console>
+        <log remove="remove"/>
+        <errorlog remove="remove"/>
+    </logger>
+
+    <query_log>
+        <database>system</database>
+        <table>query_log</table>
+    </query_log>
+
+    <timezone>Europe/Berlin</timezone>
+
+    <zookeeper>
+        <node index="1">
+            <host>%s</host>
+            <port>2181</port>
+        </node>
+    </zookeeper>
+
+    <remote_servers>
+        <default>
+            <shard>
+                <replica>
+                    <host>localhost</host>
+                    <port>9000</port>
+                </replica>
+            </shard>
+        </default>
+    </remote_servers>
+    <macros>
+        <cluster>default</cluster>
+        <shard>shard</shard>
+        <replica>replica</replica>
+    </macros>
+
+    <distributed_ddl>
+        <path>/clickhouse/task_queue/ddl</path>
+    </distributed_ddl>
+
+    <format_schema_path>/var/lib/clickhouse/format_schemas/</format_schema_path>
+</clickhouse>
+`, c.IP())
```

Comment:
this would be a limitation for those who need other configurations.

Reply:
actually, it's possible - https://clickhouse.com/docs/en/operations/configuration-files
I will put this zookeeper config in a separate file.